### PR TITLE
Guarin lig 136 add multi gpu support

### PR DESCRIPTION
**Pull request by guarin**

### CODE REVIEW

File: lightly/loss/swav_loss.py [hunk 1]
```diff
@@ -22,23 +27,33 @@ def sinkhorn(out: torch.Tensor, iterations: int = 3, epsilon: float = 0.05):
...
     K = Q.shape[0] # number of prototypes
 
-    for i in range(iterations):
+    for _ in range(iterations):
         # normalize rows
         sum_of_rows = torch.sum(Q, dim=1, keepdim=True)
+        if world_size > 1:
+            dist.all_reduce(sum_of_rows)
         Q /= sum_of_rows
         Q /= K
```
Comment (MalteEbner) [inline lightly/loss/swav_loss.py:58]: This line might be redundant.

**Review by MalteEbner — COMMENTED** [review]
Looks good to me, only some comments could be improved.

File: lightly/loss/swav_loss.py (hunk 1 above)
Comment (MalteEbner) [reply lightly/loss/swav_loss.py:58]: Q /= K is redundant, as the columns are normalised in the line below.

Comment (MalteEbner) [reply lightly/loss/swav_loss.py:58]: If you insert `Q = Q / K` into `Q = Q / torch.sum(Q, dim=0, keepdim=True)` then you have
`Q = (Q / K) / torch.sum((Q / K), dim=0, keepdim=True)` and the K cancels itself out.

Comment (guarin) [reply lightly/loss/swav_loss.py]: I agree with you :)

**Review by MalteEbner — APPROVED** [review]
LGTM
